provider/openstack/provider.go

```diff
@@ -803,11 +824,7 @@ var authenticateClient = func(auth authenticator) error {
 		// but provide a readable and helpful error message
 		// to the user.
 		logger.Debugf("authentication failed: %v", err)
-		return errors.New(`authentication failed.
-
-Please ensure the credentials are correct. A common mistake is
-to specify the wrong tenant. Use the OpenStack "project" name
-for tenant-name in your model configuration.`)
+		return errors.Errorf("authentication failed : %v", err)
```

Comment:
As discussed, we need to look at whether the error is an auth error and only print the credential error message if it is. Otherwise we print the error itself.